server/bot/commands/trivia.ts

```diff
@@ -0,0 +1,259 @@
+
+    const revealMessages = ['I know this: ', 'Easy: ', 'Of course, it\'s: ']
+    const failMessages = ['You suck at this', 'That was trivial, really', 'Moving on...']
+
+    if (this.status === 'correct answer') {
+      this.status = 'new question'
+      await new Promise(resolve => setTimeout(resolve, 1000))
+      if (this.status !== 'stop') await this.newQuestion()
+    } else if (this.status === 'stop') {
+      return true
+    } else {
+      let message: string
+      if (this.settings.revealAnswer) {
+        message = revealMessages[Math.floor(Math.random() * revealMessages.length)] + this.currentLine.answers[0]
+      } else {
+        message = failMessages[Math.floor(Math.random() * failMessages.length)]
+      }
+      if (this.settings.botPlays) {
+        message += '\n**+1** for me!'
+        if (!this.scores['Jony Ive']) {
+          this.scores['Jony Ive'] = 1
+        } else {
+          this.scores['Jony Ive'] += 1
+        }
+      }
+      this.currentLine = null
+      await this.channel.createMessage(message)
+      await this.channel.sendTyping()
+      await new Promise(resolve => setTimeout(resolve, 1000))
+      if (this.status !== 'stop') await this.newQuestion()
+    }
+  }
+
+  checkAnswer (message: Message) {
+    if (message.author.bot || this.currentLine === null) {
+      return false
+    }
+    this.timeout = Date.now()
+    let hasGuessed = false
+
+    for (let i = 0; i < this.currentLine.answers.length; i++) {
+      let answer = this.currentLine.answers[i].toLowerCase()
+      let guess = message.content.toLowerCase()
+      if (!answer.includes(' ')) { // Strict answer checking for one word answers
+        const guessWords = guess.split(' ')
+        for (let j = 0; j < guessWords.length; j++) {
+          if (guessWords[j] === answer) {
+            hasGuessed = true
+          }
+        }
+      } else if (guess.includes(answer)) { // The answer has spaces, checking isn't as strict
+        hasGuessed = true
+      }
+    }
+
+    if (hasGuessed) {
+      this.currentLine = null
+      this.status = 'correct answer'
+      if (!this.scores[message.author.username]) {
+        this.scores[message.author.username] = 1
+      } else {
+        this.scores[message.author.username] += 1
+      }
+      await this.channel.createMessage(`You got it ${message.author.username}! **+1** to you!`)
+    }
+  }
+}
+
+export const handleTrivia: Command = {
+  name: 'trivia',
+  opts: {
+    description: 'Start a trivia session',
+    fullDescription: 'Start a trivia session\nDefault settings are:\nIve gains points: false\nSeconds to answer: 15\nPoints needed to win: 30\nReveal answer on timeout: true',
+    usage: '/trivia <topic> (--bot-plays=true|false) (--time-limit=<time longer than 4s>) (--max-score=<points greater than 0>) (--reveal-answer=true|false)\n During a trivia session, the following commands may also be run:\n/trivia score\n/trivia stop',
+    example: '/trivia greekmyth --bot-plays=true',
```

Comment:
This seems to be incomplete? scores, leaderboard and list aren't there, and it also uses too many lines.